applications/DEM_application/custom_conditions/RigidFace.cpp

```diff
@@ -428,11 +428,13 @@ bool RigidFace3D::CheckProjectionFallsInside(SphericParticle *p_particle)
     const array_1d<double, 3> w  = P - geom[0].Coordinates();
     array_1d<double, 3> u1 = geom[1].Coordinates() - geom[0].Coordinates();
     array_1d<double, 3> u2 = geom[2].Coordinates() - geom[0].Coordinates();
+    array_1d<double, 3> u2_copy;
+    noalias(u2_copy) = u2;
     array_1d<double, 3> n;
-    DEM_SET_TO_CROSS_OF_FIRST_TWO_3(u1, u2, n)
-    DEM_SET_TO_CROSS_OF_FIRST_TWO_3(w, u2, u2)
+    GeometryFunctions::CrossProduct(u1, u2, n);
+    GeometryFunctions::CrossProduct(w, u2_copy, u2);
```

Comment:
just to tell that you could avoid this product since u2 x u2 must return a zero vector (u2 is parallel to itself)

Reply:
I am seeing that I forgot to remove the scheme from the strategies in the swimming DEM part. I only checked that everything compiles and the DEM tests, not the swimming DEM tests, sorry. I can try to fix it.

Reply:
Hello, I already corrected this in another pull request awaiting revision

Reply:
Hello, no, the third parameter is the result.